freetextresponse/views.py

```diff
@@ -0,0 +1,367 @@
+        """
+        result = ''
+        if (
+                (ignore_attempts or self.count_attempts > 0) and
+                (not self._word_count_valid())
+        ):
+            word_count_message = self._get_word_count_message()
+            result = ugettext(
+                "Invalid Word Count. {word_count_message}"
+            ).format(
+                word_count_message=word_count_message,
+            )
+        return result
+
+    def _get_submitted_message(self):
+        """
+        Returns the message to display in the submission-received div
+        """
+        result = ''
+        if self._word_count_valid():
+            result = self.submitted_message
+        return result
+
+    def _get_user_alert(self, ignore_attempts=False):
+        """
+        Returns the message to display in the user_alert div
+        depending on the student answer
+        """
+        result = ''
+        if not self._word_count_valid():
+            result = self._get_invalid_word_count_message(ignore_attempts)
+        return result
+
+    def _can_submit(self):
+        """
+        Determine if a user may submit a response
+        """
+        if self.is_past_due():
+            return False
+        if self.max_attempts == 0:
+            return True
+        if self.count_attempts < self.max_attempts:
+            return True
+        return False
+
+    @classmethod
+    def _generate_validation_message(cls, msg):
+        """
+        Helper method to generate a ValidationMessage from
+        the supplied string
+        """
+        result = ValidationMessage(
+            ValidationMessage.ERROR,
+            ugettext(unicode(msg))
+        )
+        return result
+
+    def validate_field_data(self, validation, data):
+        """
+        Validates settings entered by the instructor.
+        """
+        if data.weight < 0:
+            msg = FreeTextResponseViewMixin._generate_validation_message(
+                'Weight Attempts cannot be negative'
+            )
+            validation.add(msg)
+        if data.max_attempts < 0:
+            msg = FreeTextResponseViewMixin._generate_validation_message(
+                'Maximum Attempts cannot be negative'
+            )
+            validation.add(msg)
+        if data.min_word_count < 1:
+            msg = FreeTextResponseViewMixin._generate_validation_message(
+                'Minimum Word Count cannot be less than 1'
+            )
+            validation.add(msg)
+        if data.min_word_count > data.max_word_count:
+            msg = FreeTextResponseViewMixin._generate_validation_message(
+                'Minimum Word Count cannot be greater than Max Word Count'
+            )
+            validation.add(msg)
+        if not data.submitted_message:
+            msg = FreeTextResponseViewMixin._generate_validation_message(
+                'Submission Received Message cannot be blank'
+            )
+            validation.add(msg)
```

Comment:
Or maybe don't use `ugettext_lazy`, I'm not so sure.

Just bringing your attention to the issue. So, which is better in your opinion?